instagram/src/components/Authentication/Authenticate.js

```diff
@@ -1,21 +1,27 @@
 import React from 'react';
 
+import AddPostPage from '../PostContainer/AddPostPage';
+
 const Authenticate = (Page) => (Login) => {
 
   return class extends React.Component {
 
     render() {
 
       return this.props.loggedIn ?
-      (<Page
-        displayedData={this.props.displayedData}
-        searchFunc={this.props.searchFunc}
-        addLike={this.props.addLike}
-        addComment={this.props.addComment}
-        removeComment={this.props.removeComment}
-        username={this.props.username}
-        setUsername={this.props.setUsername}
-      />)
+        (this.props.addingPost ?
+        (<AddPostPage submitPost={this.props.submitPost} addPost={this.props.addPost} />)
+        :
```

Comment:
I'm a bit confused by the formatting for this ternary. I think it could use some work for uniformity! One approach might be something like this:
```javascript
return this.props.loggedIn
  ? (
    this.props.addPost
      ? (
        <AddPostPage
          submitPost={this.props.submitPost}
          addPost={this.props.addPost) 
        />
      )
      : (
        <Page
          displayedData={this.props.displayedData}
          searchFunc={this.props.searchFunc}
          addLike={this.props.addLike}
          addComment={this.props.addComment}
          removeComment={this.props.removeComment}
          username={this.props.username}
          setUsername={this.props.setUsername}
          addPost={this.props.addPost} 
        />
      )
  )
  : <Login setUsername={this.props.setUsername} />;
```
You could potentially also reduce some of the pain and repetition of passing props by using [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax):
```javascript
       <Page {...props} />
```